### PR TITLE
MobileUI: Do not show allergen circle and allow higher weight than planned when in tolerance

### PR DESCRIPTION
**Pull request by fp-ps**
https://github.com/metasfresh/metasfresh/issues/14869

### CODE REVIEW

File: misc/services/mobile-webui/mobile-webui-frontend/src/utils/qtys.js [hunk 1]
```diff
@@ -74,3 +47,43 @@ const getDefaultDisplayPrecision = (uom, defaultPrecision = 4) => {
     return defaultPrecision;
   }
 };
+
+export const computeEffectiveValues = ({ qty, uom, precision = null }) => {
```
Comment (teosarca) [inline misc/services/mobile-webui/mobile-webui-frontend/src/utils/qtys.js:51]: IMHO `computeEffectiveValues` name is quite ambiguous.

suggestion:
* rename `formatQtyToHumanReadable` to `formatQtyToHumanReadableString`
* rename `computeEffectiveValues` to `formatQtyToHumanReadable`

wdyt?

File: misc/services/mobile-webui/mobile-webui-frontend/src/components/buttons/ButtonWithIndicator.jsx [hunk 2]
```diff
@@ -22,7 +22,13 @@ const ButtonWithIndicator = ({
...
   const indicatorClassName = getIndicatorClassName(completeStatus);
 
-  const showHazardsAndAllergens = hazardSymbols != null || allergens != null;
+  const allergensWithColor = allergens != null && allergens.filter((allergen) => allergen.color != null);
+
+  const displayAllergens = allergensWithColor.length > 0;
+
+  const displayHazards = hazardSymbols != null && hazardSymbols.length > 0;
+
+  const showHazardsAndAllergens = displayHazards || displayAllergens;
```
Comment (teosarca) [inline misc/services/mobile-webui/mobile-webui-frontend/src/components/buttons/ButtonWithIndicator.jsx:31]: rename `showHazardsAndAllergens` to `displayHazardsAndAllergens` (for consistency)

File: misc/services/mobile-webui/mobile-webui-frontend/src/components/buttons/ButtonWithIndicator.jsx [hunk 3]
```diff
@@ -22,7 +22,13 @@ const ButtonWithIndicator = ({
 }) => {
   const indicatorClassName = getIndicatorClassName(completeStatus);
 
-  const showHazardsAndAllergens = hazardSymbols != null || allergens != null;
+  const allergensWithColor = allergens != null && allergens.filter((allergen) => allergen.color != null);
+
+  const displayAllergens = allergensWithColor.length > 0;
```
Comment (teosarca) [inline misc/services/mobile-webui/mobile-webui-frontend/src/components/buttons/ButtonWithIndicator.jsx:27]: not sure, but i think u shall do
```javascript
const displayAllergens = allergensWithColor && allergensWithColor.length > 0;
```

File: misc/services/mobile-webui/mobile-webui-frontend/src/utils/qtys.js (hunk 1 above)
Comment (fp-ps) [reply misc/services/mobile-webui/mobile-webui-frontend/src/utils/qtys.js:51]: @teosarca the renaming sounds good. on it

**Review by teosarca — APPROVED** [review]

**Pull request merged**